crawl-ref/source/describe.cc

```diff
@@ -3884,50 +3884,50 @@ static void _add_energy_to_string(int speed, int energy, string what,
 }
 
 /**
- * Calculate effects on the player's to-hit from a defending monster.
+ * Calculate post-randomization effects on the player's to-hit.
  *
- * @param base_to_hit   The to-hit with only attacker-side effects applied.
+ * @param to_hit        The actual randomized to-hit.
  * @param mi[in]        Player-visible info about the monster in question.
  * @param melee         Whether the attack is melee or ranged.
  */
-static int _apply_defender_effects(const monster_info& mi, int to_hit, bool melee)
+static int _post_roll_to_hit_modifiers(const monster_info& mi, int to_hit, bool melee)
 {
     int effect = 0;
-    // Lighting effects. Subtracting these from EV is weirdly equivalent to adding
-    // them to the post-roll to-hit (the to_hit we have here is pre-roll).
+
+    effect -= 5 * you.inaccuracy();
+
+    if (you.confused()) effect -= 5;
```

Comment:
It seems like we can factor out all of the post-roll effects that don't depend on the defender to methods on attack, melee_attack, and ranged_attack, right? (I can do this myself after merging your PR into mine, if you prefer.)

Then we'd only have lighting effects and rmsl here.

Reply:
Sure, that works. I'll let you handle that. :)